docker/docker-prepare.sh

```diff
@@ -89,6 +89,46 @@ superuser() {
 	fi
 }
 
+custom_container_init() {
+	# Mostly borrowed from the LinuxServer.io base image
+	# https://github.com/linuxserver/docker-baseimage-ubuntu/tree/bionic/root/etc/cont-init.d
+	readonly custom_script_dir="/custom-cont-init.d"
+	# Tamper checking.
+	# Don't run files which are owned by anyone except root
+	# Don't run files which are writeable by others
+	if [ -d "${custom_script_dir}" ]; then
+		if [ -n "$(find "${custom_script_dir}" ! -user root)" ]; then
```

Comment:
I tested this out, and it will also pick up if the top folder is not root owned,  I did limit the depth to 1 though, since only things directly inside the folder matter/could be run